### PR TITLE
Correct self attribute in ATOM feeds

### PR DESCRIPTION
**Pull request by Alkarex**
Fix for https://github.com/friendica/friendica/issues/6128#issuecomment-439016471
Refactoring of https://github.com/friendica/friendica/pull/6140

### CODE REVIEW

File: src/Protocol/OStatus.php [hunk 1]
```diff
@@ -1283,10 +1284,19 @@ private static function addHeader(DOMDocument $doc, array $owner, $filter)
 		$root->setAttribute("xmlns:mastodon", NAMESPACE_MASTODON);
 
 		$title = '';
+		$selfUri = '/feed/' . $owner["nick"] . '/';
```
Comment (annando) [inline src/Protocol/OStatus.php]: I would prefer having all ```$selfUri``` assignments at a single place.

Comment (Alkarex) [reply src/Protocol/OStatus.php]: Like repeating this file for each of the switch cases?

Comment (Alkarex) [reply src/Protocol/OStatus.php]: @annando New proposition https://github.com/friendica/friendica/pull/6144/commits/bd19e93c094b85bc2fb40044d972dcb98ce53d7c

Comment (annando) [reply src/Protocol/OStatus.php]: Why not simply:
```
if (!$feed_mode) {
	$selfUri = "/dfrn_poll/" . $owner["nick"];
} else {
	$selfUri** = '/feed/' . $owner["nick"] . '/';
}
```

Comment (Alkarex) [reply src/Protocol/OStatus.php]: @annando Your code does not seem to work for for the 2 other types of feeds, does it?
```
/feed/nick/comments
/feed/nick/activity
```

Comment (annando) [reply src/Protocol/OStatus.php]: Ah, thanks!

**Pull request merged**